platform/mbed_poll.cpp

```diff
@@ -0,0 +1,75 @@
+/* mbed Microcontroller Library
+ * Copyright (c) 2017 ARM Limited
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+#include "mbed_poll.h"
+#include "FileHandle.h"
+#include "Timer.h"
+#ifdef MBED_CONF_RTOS_PRESENT
+#include "rtos/Thread.h"
+#endif
+
+namespace mbed {
+
+// timeout -1 forever, or milliseconds
+int poll(pollfh fhs[], unsigned nfhs, int timeout)
+{
+    // Quick initial hack that spins
```

Comment:
Because we were hoping a proper wake up solution will be available for mbed-os, like Conditional variables. Unfortunately, we couldn't get it. Russ do have an implementation for that but he didn't put it through or something. Currently poll() is blocking because of this lack of wakeup mechanism.